src/screens/Asl/components/AslTable/index.js

```diff
@@ -0,0 +1,236 @@
+    const [sortConfig, setSortConfig] = useState(config);
+    
+  
+    const sortedItems = useMemo(() => {
+      let sortableItems = [...items];
+      if (sortConfig !== null) {
+        sortableItems.sort((a, b) => {
+          if (a[sortConfig.key] < b[sortConfig.key]) {
+            return sortConfig.direction === "ascending" ? -1 : 1;
+          }
+          if (a[sortConfig.key] > b[sortConfig.key]) {
+            return sortConfig.direction === "ascending" ? 1 : -1;
+          }
+          return 0;
+        });
+      }
+      return sortableItems;
+    }, [items, sortConfig]);
+  
+    const requestSort = key => {
+      let direction = "ascending";
+      if (sortConfig && 
+        sortConfig.key === key && 
+        sortConfig.direction === 'ascending') {
+            direction = 'descending';
+        }
+      setSortConfig({ key, direction });
+    };
+  
+    return { items:sortedItems, requestSort, sortConfig};
+};
+
+
+
+const AslTable = props => {
+    const ONE_PAGE_NUM = 50 // one page will have at most 50 glossaries
+    const { items, requestSort, sortConfig } = useSortableData(props.words);
+    const [pageNumber, setPageNumber] = useState(1); // the page number is at first 1
+    const [jumpNumber, setJumpNumber] = useState(1);
+    const [length, setLength] = useState(0); // the length of filtered items is set to 0
+    const [search, setSearch] = useState(''); // search text is at first empty
+    const [onePage, setOnePage] = useState([]);
+
+    const apiInstance = axios.create({
+      baseURL: 'https://ct-dev.ncsa.illinois.edu',
+      timeout: 1000,
+    });
+
+    useEffect(() => {
+      const index = (pageNumber-1) * ONE_PAGE_NUM;
+      setOnePage(items.filter(item => item.term.toLowerCase().includes(search.toLowerCase())).slice(index, index + ONE_PAGE_NUM));
+    }, [pageNumber]); // we set page after pagenumber changed
+
+    useEffect(() => {
+      const filtered = items.filter(item => item.term.toLowerCase().includes(search.toLowerCase()));
+      setLength(filtered.length);
+      setPageNumber(1);
+      setOnePage(filtered.slice(0, 50));
+    }, [search])
+
+    useEffect(() => {
+      setLength(items.length);
+      setSearch('');
+      setPageNumber(1);
+      setOnePage(items.slice(0, 50));
+    }, [items])
+
+    const getClassNamesFor = name => {
+        if (!sortConfig) {
+          return;
+        }
+        return sortConfig.key === name ? sortConfig.direction : undefined;
+    };
+    
+    const handleJump = () => {
+      if (jumpNumber >= 1 && jumpNumber*ONE_PAGE_NUM < length) {
+        // pagenumber is valid
+        setPageNumber(jumpNumber);
+      } else {
+        alert("please enter a valid page number!");
+      }
+    }
+
+    const numberToCate = (num) => {
+      if (num === 1) {
+        return 'Sign'
+      }
+      if (num === 2) {
+        return 'Definition'
+      }
+      return 'Example'
+    }
+
+    const getVideoLink = (source, uniqueASLIdentifier) => {
```

Comment:
@lijiaxi2018 This logic to convert a uniqueASLIdentifier to the mp4 link for different sources should not be in the frontend. Presumably this should be a relative link stored in the database, like the other videos.

Reply:
If I understand correctly, the video link will just be uniqueASLIdentifier, right?

Reply:
@lijiaxi2018 Jiaxi, a better design might give the mp4 link to the frontend? (Should we store it as DB column)